x-pack/plugins/observability/public/utils/observability_data_views/observability_data_views.ts

```diff
@@ -230,6 +231,9 @@ export class ObservabilityDataViews {
         if (e instanceof SavedObjectNotFound) {
           return await this.createAndSavedDataView(app, appIndices);
         }
+        if (e instanceof DataViewMissingIndices) {
+          this.dataViews.clearInstanceCache();
```

Comment:
Instead of clearing the cache you could refresh the field list for a given data view - `dataViews.refreshFields( dataView)`

Reply:
that sounds better. let me try that.

Reply:
Can you pass the data view id to `clearInstanceCache`?